sample/src/main/java/com/example/CashAppV3SampleActivity.kt

```diff
@@ -0,0 +1,284 @@
+import app.cash.paykit.core.CashAppPay
+import app.cash.paykit.core.CashAppPayFactory
+import app.cash.paykit.core.CashAppPayListener
+import app.cash.paykit.core.CashAppPayState
+import app.cash.paykit.core.CashAppPayState.Approved
+import app.cash.paykit.core.CashAppPayState.Authorizing
+import app.cash.paykit.core.CashAppPayState.CashAppPayExceptionState
+import app.cash.paykit.core.CashAppPayState.CreatingCustomerRequest
+import app.cash.paykit.core.CashAppPayState.Declined
+import app.cash.paykit.core.CashAppPayState.NotStarted
+import app.cash.paykit.core.CashAppPayState.PollingTransactionStatus
+import app.cash.paykit.core.CashAppPayState.ReadyToAuthorize
+import app.cash.paykit.core.CashAppPayState.Refreshing
+import app.cash.paykit.core.CashAppPayState.RetrievingExistingCustomerRequest
+import app.cash.paykit.core.CashAppPayState.UpdatingCustomerRequest
+import app.cash.paykit.core.models.sdk.CashAppPayCurrency.USD
+import app.cash.paykit.core.models.sdk.CashAppPayPaymentAction
+import com.afterpay.android.Afterpay
+import com.afterpay.android.model.CheckoutV3CashAppPay
+import com.afterpay.android.model.CheckoutV3Data
+import com.afterpay.android.model.Configuration
+import com.example.databinding.CashAppV3LayoutBinding
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+
+/**
+ * Activity showing the Cash App Pay V3 checkout flow
+ */
+class CashAppV3SampleActivity : AppCompatActivity() {
+
+    private lateinit var bindings: CashAppV3LayoutBinding
+
+    private val cashAppPay: CashAppPay =
+        CashAppPayFactory.createSandbox(AFTERPAY_ENVIRONMENT.payKitClientId)
+
+    private var checkoutV3CashAppPay: CheckoutV3CashAppPay? = null
+
+    private val cashAppPayListener =
+        object : CashAppPayListener {
+            override fun cashAppPayStateDidChange(newState: CashAppPayState) {
+                Log.d(tag, "cashAppPayStateDidChange: ${newState::class.java}")
+                when (newState) {
+                    is ReadyToAuthorize -> {
+                        /**
+                         * Step 5: Cash App Pay SDK will response when an authorization attempt is ready
+                         * We can now enable the button to let customer proceed with checkout
+                         */
+                        lifecycleScope.launch { // jump back to UI thread to update UI
+                            bindings.cashappPayButton.apply {
+                                isEnabled = true
+                                setOnClickListener { _ ->
+                                    /**
+                                     * Step 6: When customer clicks Cash App Pay button
+                                     *  begin authorization. This will begin UI flow into Cash App
+                                     */
+                                    cashAppPay.authorizeCustomerRequest()
+                                }
```

Comment:
Nit: because the short TTL on the "Created Customer Request", it will auto-refresh every 30s or every minute, and what happens is that the state temporarily transitions into `Refreshing` and then back to `ReadyToAuthorize` . So there is a possibility we end up calling into this `setOnClickListener` multiple times. It's not wrong, but seems wasteful. So ideally we would put this elsewhere; the `isEnabled` makes sense to be here however 👍🏻